src/scala/ly/stealth/mesos/kafka/Constraint.scala

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ly.stealth.mesos.kafka
+
+import java.util.regex.PatternSyntaxException
+
+class Constraint(_value: String) {
+  var _condition: Constraint.Condition = null
+
+  def value: String = _value
+  def condition: Constraint.Condition = _condition
+
+  this.parse
+  private def parse: Unit = {
+    if (_value.startsWith("like:")) _condition = new Constraint.Like(_value.substring("like:".length))
```

Comment:
Thanks for a suggestion. Not sure about it, cause some branches contains prefix matching.